daemon/cmd/daemon_main.go

```diff
@@ -1279,6 +1279,10 @@ func initEnv(vp *viper.Viper) {
 
 	linuxdatapath.CheckRequirements()
 
+	if err := probes.CreateHeaderFiles(filepath.Join(option.Config.BpfDir, "include/bpf"), probes.ExecuteHeaderProbes()); err != nil {
+		log.WithError(err).Fatal("failed to create header files with feature macros")
+	}
+
```

Comment:
Should we continue to respect `option.Config.DryMode` here?

Reply:
I don't think that is necessary unless we see failing tests that would expect `CreateHeaderFiles` to be guarded by `DryMode`. Since we are moving it to `initEnv` which already does a bunch of other probing without respecting `DryMode`, this would just create unnecessarily nested code.